python/torch_mlir/eager_mode/torch_mlir_dispatch.py

```diff
@@ -0,0 +1,272 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+from typing import Any, Callable, Tuple, Union
+from typing import List, Dict
+
+import numpy as np
+import torch
+import torch._C
+from torch.fx.node import map_aggregate
+from torch.fx.operator_schemas import normalize_function, create_type_hint
+from torch.utils._pytree import tree_map
+from torch_mlir._mlir_libs._mlir.passmanager import PassManager
+
+from torch_mlir.dialects import torch as torch_dialect
+from torch_mlir.dialects.torch.importer.jit_ir.build_tools.registry import (
+    Registry,
+    JitOperator,
+)
+from torch_mlir.eager_mode.ir_building import build_module, TorchTensorType
+
+OP_REGISTRY = Registry.load()
+SUPPORTED_OPS = frozenset(
+    [
+        member.OPERATION_NAME
+        for member in vars(torch_dialect).values()
+        if hasattr(member, "OPERATION_NAME")
+    ]
+)
+
+
+class UnsupportedByTorchMlirEagerMode(Exception):
+    def __init__(self, value: str):
+        super().__init__()
+        self.value = value
+
+    def __str__(self) -> str:
+        return self.value
+
+
+def check_supported_op(schema: torch._C.FunctionSchema) -> bool:
+    return (
+        "torch."
+        + schema.name.replace("::", ".")
+        + ("." + schema.overload_name if schema.overload_name else "")
+    ) in SUPPORTED_OPS
+
+
+def is_tensor_type(typ: torch._C.Type):
+    return typ.isSubtypeOf(torch.TensorType.get()) or (
+        isinstance(typ, torch.OptionalType)
+        and typ.getElementType().isSubtypeOf(torch._C.TensorType.get())
+    )
+
+
+def normalize_args_kwargs(target: Callable, args: Tuple[Any], kwargs: Dict[str, Any]):
+    """
+    Primarily for filling in default values for optional args, which are dependent on the schema.
+    """
+    arg_types = map_aggregate(args, type)
+    assert isinstance(arg_types, tuple)
+    arg_types = tuple([create_type_hint(i) for i in arg_types])
+    kwarg_types = {k: type(v) for k, v in kwargs.items()}
+
+    new_args_and_kwargs = normalize_function(
+        target, args, kwargs, arg_types, kwarg_types, normalize_to_only_use_kwargs=False
+    )
+    assert new_args_and_kwargs, "Couldn't normalize args and kwargs"
+    new_args, new_kwargs = new_args_and_kwargs
+    return new_args, new_kwargs
+
+
+def build_script_function(
+    schema: torch._C.FunctionSchema,
+    args: List[torch._C.Argument],
+    kwargs: Dict[str, Any],
+) -> torch.jit.ScriptFunction:
+    """
```

Comment:
Nit: elsewhere in Torch-MLIR, we always have a one-line summary on the same line as the opening `"""`. Can you update the docstrings to match that convention?